internal/impl/nats/integration_kv_test.go

```diff
@@ -0,0 +1,319 @@
+package nats
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/nats-io/nats.go"
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/benthosdev/benthos/v4/internal/integration"
+	"github.com/benthosdev/benthos/v4/public/service"
+)
+
+func TestIntegrationNatsKV(t *testing.T) {
+	integration.CheckSkip(t)
+	t.Parallel()
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	pool.MaxWait = time.Second * 30
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "nats",
+		Tag:        "latest",
+		Cmd:        []string{"--js", "--trace"},
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		assert.NoError(t, pool.Purge(resource))
+	})
+
+	var natsConn *nats.Conn
+	_ = resource.Expire(900)
+	require.NoError(t, pool.Retry(func() error {
+		natsConn, err = nats.Connect(fmt.Sprintf("tcp://localhost:%v", resource.GetPort("4222/tcp")))
+		return err
+	}))
+	t.Cleanup(func() {
+		natsConn.Close()
+	})
+
+	template := `
+output:
+  label: kv_output
+  nats_kv:
+    urls: [ tcp://localhost:$PORT ]
+    bucket: bucket-$ID
+    # We need to make this key random as the NATS server will only deliver the
+    # latest revision of a key when it's requested by a watcher, this is by
+    # design, but if we want to test benthos semantics like batching we should
+    # use unique keys for ever message passing through the output
```

Comment:
ERRRRRMMMMMMMM... I _think_ you mean *__every__* message...

What I disaster :sweat: